### ISSUE TITLE
`inCluster` config is broken for IPv6

### ISSUE THREAD

**Issue by rustycl0ck** [dexidp/dex#1643]
https://github.com/dexidp/dex/blob/a901e2f2048de70c30df49781a4fbb9aa3413f4f/storage/kubernetes/client.go#L419

This breaks for IPv6 clusters with error:
`failed to list custom resource authcodes.dex.coreos.com, attempting to create: Get https://dead:beef::1:443/apis/dex.coreos.com/v1/namespaces/auth/authcodes: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

**Comment by JerrySunWRS** [comment]
I am working in the Openstack StarlingX team and recently encountered the same issue. I was able to fix it by applying the following change:
https://opendev.org/starlingx/oidc-auth-armada-app/src/branch/master/dex/centos/docker_patches/0001-Wrap-Kubernetes-host-address-in-square-brackets.patch
We were planning to open an issue here and propose the fix, but we did not get around to it until now. Since you have opened an issue already, I thought this would be the right place to mention the fix.
